WordPress/Classes/Utility/DiskCache.swift

```diff
@@ -0,0 +1,273 @@
+        guard !items.isEmpty else {
+            return
+        }
+        var size = items.reduce(0) { $0 + ($1.meta.totalFileAllocatedSize ?? 0) }
+
+        guard size > configuration.sizeLimit else {
+            return // All good, no need to perform any work.
+        }
+
+        let targetSizeLimit = Int(Double(configuration.sizeLimit) * configuration.trimRatio)
+
+        // Most recently accessed items first
+        let past = Date.distantPast
+        items.sort { // Sort in place
+            ($0.meta.contentAccessDate ?? past) > ($1.meta.contentAccessDate ?? past)
+        }
+
+        // Remove the items until it satisfies both size and count limits.
+        while size > targetSizeLimit, let item = items.popLast() {
+            size -= (item.meta.totalFileAllocatedSize ?? 0)
+            try? FileManager.default.removeItem(at: item.url)
+        }
+    }
+
+    // MARK: Contents
+
+    struct Entry {
+        let url: URL
+        let meta: URLResourceValues
+    }
+
+    func contents(keys: [URLResourceKey] = []) -> [Entry] {
+        guard let urls = try? FileManager.default.contentsOfDirectory(at: rootURL, includingPropertiesForKeys: keys, options: .skipsHiddenFiles) else {
+            return []
+        }
+        let keys = Set(keys)
+        return urls.compactMap {
+            guard let meta = try? $0.resourceValues(forKeys: keys) else {
+                return nil
+            }
+            return Entry(url: $0, meta: meta)
+        }
+    }
+
+    // MARK: Inspection
+
+    /// The total number of items in the cache.
+    ///
+    /// - important: Requires disk IO, avoid using from the main thread.
+    public var totalCount: Int {
+        contents().count
+    }
+
+    /// The total file size of items written on disk.
+    ///
+    /// Uses `URLResourceKey.fileSizeKey` to calculate the size of each entry.
+    /// The total allocated size (see `totalAllocatedSize`. on disk might
+    /// actually be bigger.
+    ///
+    /// - important: Requires disk IO, avoid using from the main thread.
+    public var totalSize: Int {
+        contents(keys: [.fileSizeKey]).reduce(0) {
+            $0 + ($1.meta.fileSize ?? 0)
+        }
+    }
+
+    /// The total file allocated size of all the items written on disk.
+    ///
+    /// Uses `URLResourceKey.totalFileAllocatedSizeKey`.
+    ///
+    /// - important: Requires disk IO, avoid using from the main thread.
+    public var totalAllocatedSize: Int {
+        contents(keys: [.totalFileAllocatedSizeKey]).reduce(0) {
+            $0 + ($1.meta.totalFileAllocatedSize ?? 0)
+        }
+    }
+}
+
+private extension String {
+    var sha1: String? {
+        guard !isEmpty, let data = self.data(using: .utf8) else {
+            return nil
+        }
+        // SHA1 offers a good balance between performance and size. Git uses
+        // SHA1 for commit hashes and other purposes.
+        return Insecure.SHA1.hash(data: data)
```

Comment:
Modern iOS devices can do modern crypto very fast, so we probably don't need to use a deprecated hashing method.

That said – do we need to hash keys at all? Could we just store items on-disk using their developer-specified `key`?

Reply:
> Modern iOS devices can do modern crypto very fast, so we probably don't need to use a deprecated hashing method.

SHA1 is still significantly faster than variants that create longer keys, but not in absolute numbers. In absolute numbers, it's fast enough to be ignored.

> That said – do we need to hash keys at all? Could we just store items on-disk using their developer-specified key?

In the current filed-based implementation, yes, because many characters can't or shouldn't be used in filenames.

I say current because I'm considering closing this PR since there seems to be a lot of questions and implementing something more conservative: no actors and likely using SQLite for storage like `URLCache` – it'll be more efficient both in terms of the performance and space usage than a file-based solution for the purpose of storing small API responses.

It will also allow for arbitrary keys to be used as strings and help address your comment on `performAndScheduleNextSweep`. With a database, I'll be able to avoid any inefficient crawling of the filesystem.

Reply:
Updated it to SHA256.

SQLite - never mind, too complicated for this case for too little benefit.